src/snyk/common/configuration/trustedFolders.ts

```diff
@@ -1,7 +1,13 @@
 import { IConfiguration } from './configuration';
 
+import path from 'path';
+
 export function getTrustedFolders(config: IConfiguration, workspaceFolders: string[]): string[] {
   const trustedFolders = config.getTrustedFolders();
-
-  return workspaceFolders.filter(folder => trustedFolders.includes(folder));
+  return workspaceFolders.filter(folder => {
+    return trustedFolders.some(trustedFolder => {
+      const relative = path.relative(trustedFolder, folder);
+      return relative === '' || (relative && !relative.startsWith('..' + path.sep) && !path.isAbsolute(relative));
```

Comment:
What does the second part of the logical OR condition (`relative && !relative.startsWith('..' + path.sep) && !path.isAbsolute(relative)`) account for?

Reply:
It looks if the relative path contains `../`. If it does, it's not a sub directory. Also, if the path is absolute, the child dir is not relative to the parent dir.

Reply:
Ah, I've misinterpreted `path.relative` return. Somehow thought that the first part of OR will tell us if it's a subdirectory already.